repos/system_upgrade/el7toel8/actors/initramfs/targetinitramfsgenerator/libraries/targetinitramfsgenerator.py

```diff
@@ -0,0 +1,78 @@
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import InitrdIncludes, InstalledTargetKernelVersion, TargetInitramfsTasks
+from leapp.utils.deprecation import suppress_deprecation
+
+
+DRACUT_DIR = '/usr/lib/dracut/modules.d/'
+
+
+def copy_dracut_modules(modules):
+    """
+    Copy every dracut module with specified path into the expected directory.
+
+    original content is overwritten if exists
+    """
+    # FIXME: use just python functions instead of shell cmds
+    for module in modules:
+        if not module.module_path:
+            continue
+        try:
+            # context.copytree_to(module.module_path, os.path.join(DRACUT_DIR, os.path.basename(module.module_path)))
+            run(['cp', '-f', '-a', module.module_path, DRACUT_DIR])
+        except CalledProcessError as e:
+            api.current_logger().error('Failed to copy dracut module "{name}" from "{source}" to "{target}"'.format(
+                name=module.name, source=module.module_path, target=DRACUT_DIR), exc_info=True)
+            # FIXME: really do we want to raise the error and stop execution completely??....
```

Comment:
@Rezney  thing is, whether it's even possible to boot if this fails. E.g. in case you expect a dracut module to be e.g. used because of the storage drivers, it's possible you will not be able to boot even. Or networking could be broken. But sometimes it could be just "less crucial" things, that could made system not working completely as expected, but still bootable so you could fix it over ssh. Currently I've decided to be consistent and provide just expected targetinitramfs or "emergency console" (not sure whether it will endup with emergency console tbh; just raising the error for sure).

Reply:
ack, makes sense...